source/common/http/filter/ratelimit.cc

```diff
@@ -11,8 +11,9 @@
 namespace Http {
 namespace RateLimit {
 
-const Http::HeaderMapPtr Filter::TOO_MANY_REQUESTS_HEADER{new Http::HeaderMapImpl{
-    {Http::Headers::get().Status, std::to_string(enumToInt(Code::TooManyRequests))}}};
+const std::unique_ptr<const Http::HeaderMap> Filter::TOO_MANY_REQUESTS_HEADER{
+    new Http::HeaderMapImpl{
```

Comment:
There's also the concern about non-deterministic ordering of initialization. Not really a concern for simple local objects like this. 

I think that when done sanely (e.g. don't initialize static data that depends on other static data), and given we have proper thread joining on exit, it's probably fine.

It would make sense to document this in https://github.com/lyft/envoy/blob/master/CONTRIBUTING.md though.